source/server/http/admin.cc

```diff
@@ -753,8 +753,13 @@ Http::Code AdminImpl::handlerPrometheusStats(absl::string_view path_and_query, H
                                              Buffer::Instance& response, AdminStream&) {
   const Http::Utility::QueryParams params = Http::Utility::parseQueryString(path_and_query);
   const bool used_only = params.find("usedonly") != params.end();
+  const absl::optional<std::regex> regex =
+      (params.find("filter") != params.end())
+          ? absl::optional<std::regex>{std::regex(params.at("filter"))}
```

Comment:
I'd probably split this up into multiple lines for the sake of readability

Reply:
Drive by, this is already computed above in `handlerStats()`. I would just pass that computation to this function.

Reply:
@mattklein123 I was actually looking at https://github.com/envoyproxy/envoy/blob/89a82de7a8368cd93b1255dfb3818f497330faf2/source/server/http/admin.cc#L1208 and seems like there is no good way unless we want to change the definition of MAKE_ADMIN_HANDLER. It is when the method is called from top-level api handler.

Reply:
I see, OK. Maybe just make a smaller helper method?

Reply:
Helper method added.